File: src/main/java/co/utb/softeng/moviesapp/services/MovieService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.utb.softeng.moviesapp.services;

import co.utb.softeng.moviesapp.entities.Movie;
import java.util.List;

/**
 *
 * @author William
 */
public interface MovieService {
    
    public List<Movie> getAllMovies();
    public Movie getMovieById(Long id);
    public List<Movie> getMovieByName(String name);
    public List<Movie> getMovieByActorId(Long actorId);
    public List<Movie> getMovieByActorName(String actorName);
    public void saveOrUpdateMovie(Movie movie);
    public void deleteMovie(Long id);
    
    
}

File: src/main/webapp/resources/assets/js/app/services/movieService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
var servicesModule = angular.module('springMoviesAppServices');

servicesModule.factory('movieService', ['$http', function ($http) {
        
        return {
            apiUrl: apiUrl,
            getAllMovies: function () {
                return $http.get(this.apiUrl+'movie/');
            },
            getMovieById: function (movieId) {
                return $http.get(this.apiUrl+'movie/'+movieId);
            },
            createMovie: function (movie) {
                return $http.put(this.apiUrl+'movie/', movie);
            }
        };
    }]);

